### PR TITLE
Use sysName instead of hostname on AJAX search and new billing form

### PR DESCRIPTION
**Pull request by kedare**
DO NOT DELETE THIS TEXT

PR to display sysName instead of hostname when IP are used to add the devices. (so hostname==ip)

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by PipoCanaja** [comment]
Bonjour @kedare 
You cannot just change from hostname to sysName. Instead, you should call the function that actually display whatever is best according to user config parameter.  ( ``` $config['force_ip_to_sysname'] ``` ). 
Function is ``` format_hostname($device) ```. Any direct call to hostname should be replaced by this function.

File: includes/html/modal/new_bill.inc.php [hunk 1]
```diff
@@ -44,10 +44,10 @@
                         <select class="form-control input-sm" id="device" name="device" onchange="getInterfaceList(this)">
                             <option value=''>Select a device</option>
                             <?php
-                              $devices = dbFetchRows('SELECT * FROM `devices` ORDER BY hostname');
+                              $devices = dbFetchRows('SELECT * FROM `devices` ORDER BY sysName');
                             foreach ($devices as $device) {
                                 $selected = $device['device_id'] == $port_device_id ? " selected" : "";
-                                echo "<option value='${device['device_id']}' $selected>${device['hostname']}</option>\n";
+                                echo "<option value='${device['device_id']}' $selected>${device['sysName']}</option>\n";
```
Comment (PipoCanaja) [inline includes/html/modal/new_bill.inc.php]: Please replace direct calls to the hostname element by format_hostname($device)

**Comment by kedare** [comment]
Oh great, I was not aware there was such function and I have IP everywhere in my setup so no much ways to test, I'll make the change

**Comment by kedare** [comment]
In the case of the ajax_search thing, then should the attribute name still be hostname like before but like this ? 
``` bgp
                    $bgp[] = array(
                        'count'       => count($results),
                        'url'         => generate_peer_url($result),
                        'name'        => $name,
                        'description' => $description,
                        'localas'     => $localas,
                        'bgp_image'   => $bgp_image,
                        'remoteas'    => $remoteas,
                        'colours'     => $port_colour,
                        'hostname'    => format_hostname($result),
                        'sysName'    =>  $result['sysName']
                    );
```

**Comment by kedare** [comment]
Like this ?

**Comment by PipoCanaja** [comment]
It looks good, need to be tested now.

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/sysname-graph-title-instead-of-ip-address/10213/8

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-59-release-changelog-december-2019/10558/1

**Pull request merged**